source/common/upstream/upstream_impl.cc

```diff
@@ -631,6 +631,30 @@ ClusterInfoImpl::ClusterInfoImpl(const envoy::api::v2::Cluster& config,
     eds_service_name_ = config.eds_cluster_config().service_name();
   }
 
+  if (!config.has_cluster_type()) {
+    switch (config.type()) {
+    case envoy::api::v2::Cluster::STATIC:
+      cluster_type_.set_name(Extensions::Clusters::ClusterTypes::get().Static);
```

Comment:
Why is this needed? Perhaps just return an optional if you need access to custom cluster data and remove this switch statement?